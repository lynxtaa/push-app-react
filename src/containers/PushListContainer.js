import React, { useState, useCallback } from 'react'
import PropTypes from 'prop-types'

import PushList from '@components/PushList'
import PushListItem from '@components/PushListItem'

const PushListContainer = ({ sets }) => {
	const [hidden, setHidden] = useState([])
	const [prevSets, setPrevSets] = useState()

	const makeHandleClick = useCallback(
		id => () => setHidden(hidden => [...hidden, id]),
		[],
	)

	if (sets !== prevSets) {
		setPrevSets(sets)
		setHidden([])
	}

	return (
		<PushList>
			{sets.map(({ id, set }) => (
				<PushListItem
					key={id}
					id={id}
					className={hidden.includes(id) ? 'hide' : undefined}
					onClick={makeHandleClick(id)}
				>
					{set}
				</PushListItem>
			))}
		</PushList>
	)
}

PushListContainer.propTypes = {
	sets: PropTypes.arrayOf(
		PropTypes.shape({
			id: PropTypes.string.isRequired,
			set: PropTypes.number.isRequired,
		}),
	).isRequired,
}

export default PushListContainer
