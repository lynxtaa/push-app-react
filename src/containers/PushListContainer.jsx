import React from 'react'
import PropTypes from 'prop-types'

import PushList from '@components/PushList'
import PushListItem from '@components/PushListItem'

class PushListContainer extends React.Component {
	state = { hidden: [] }

	componentWillReceiveProps({ sets }) {
		if (sets !== this.props.sets) {
			this.setState({ hidden: [] })
		}
	}

	hideSet = id => this.setState(
		({ hidden }) => ({ hidden: [...hidden, id] })
	)

	render() {
		return (
			<PushList>
				{this.props.sets.map(({ id, set }) => (
					<PushListItem
						key={id}
						id={id}
						className={this.state.hidden.includes(id) ? 'hide' : undefined}
						onClick={this.hideSet}
					>
						{set}
					</PushListItem>
				))}
			</PushList>
		)
	}
}

PushListContainer.propTypes = {
	sets: PropTypes.arrayOf(PropTypes.shape({
		id: PropTypes.string.isRequired,
		set: PropTypes.number.isRequired,
	})).isRequired,
}

export default PushListContainer
