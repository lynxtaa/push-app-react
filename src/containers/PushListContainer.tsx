import { List } from '@chakra-ui/core'
import React, { useState, useEffect } from 'react'

import PushListItem from 'components/PushListItem'

interface Set {
	id: string
	set: number
}

interface Props {
	sets: Set[]
}

export default function PushListContainer({ sets }: Props) {
	const [hidden, setHidden] = useState<string[]>([])

	useEffect(() => {
		setHidden([])
	}, [sets])

	return (
		<List>
			{sets.map(({ set, id }) => (
				<PushListItem
					key={id}
					onClick={() => setHidden([...hidden, id])}
					isHidden={hidden.includes(id)}
				>
					{set}
				</PushListItem>
			))}
		</List>
	)
}
