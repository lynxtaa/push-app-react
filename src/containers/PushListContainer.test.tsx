import React from 'react'
import { render, fireEvent } from '@testing-library/react'
import createMockRaf, { MockRaf } from '@react-spring/mock-raf'
import { Globals, FrameLoop } from 'react-spring'

import PushListContainer from './PushListContainer'

let mockRaf: MockRaf

beforeEach(() => {
	mockRaf = createMockRaf()

	Globals.assign({
		now: mockRaf.now,
		requestAnimationFrame: mockRaf.raf,
		cancelAnimationFrame: mockRaf.cancel,
		frameLoop: new FrameLoop(),
	})
})

const sets = [
	{ id: '1', set: 1 },
	{ id: '2', set: 2 },
	{ id: '3', set: 3 },
]

it('renders list', () => {
	const { container } = render(<PushListContainer sets={sets} />)
	expect(container.firstChild).toMatchSnapshot()
})

it('hides item after click', async () => {
	const { getByText } = render(<PushListContainer sets={sets} />)
	const item = getByText('2')
	fireEvent.click(item)
	mockRaf.flush()
	expect(item).not.toBeVisible()
})
