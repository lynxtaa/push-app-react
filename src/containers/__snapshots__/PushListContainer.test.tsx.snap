// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders list 1`] = `
<ul
  class="pushlist list-unstyled"
>
  <li
    style="opacity: 0; margin-bottom: 0px; height: 0px; visibility: hidden;"
  >
    <button
      class="btn btn-outline-secondary w-100"
      type="button"
    >
      1
    </button>
  </li>
  <li
    style="opacity: 0; margin-bottom: 0px; height: 0px; visibility: hidden;"
  >
    <button
      class="btn btn-outline-secondary w-100"
      type="button"
    >
      2
    </button>
  </li>
  <li
    style="opacity: 0; margin-bottom: 0px; height: 0px; visibility: hidden;"
  >
    <button
      class="btn btn-outline-secondary w-100"
      type="button"
    >
      3
    </button>
  </li>
</ul>
`;
